examples/run_examples.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# set -e
```

Comment:
Because it causes the script to fail on the first error, rather than letting it continue to try all of the examples first.

Reply:
That's not correct. Neither of
```sh
if false; then echo NOPE; else echo YUP; fi
```
and
```sh
false || echo "failed"
```
cause a `set -e` script to abort.

Reply:
Because `set -e` causes the shell script to exit when any non-zero return codes from commands are encountered, and doesn't have anything to do with parsing logic inside your script.

Reply:
> Because set -e causes the shell script to exit when any non-zero return codes from commands are encountered,

I don't think that's true. Counterexample:
```sh
$ cat xy.sh
#! /bin/bash

set -e

if false; then
        echo NOPE
else 
        echo YUP
fi
$ ./xy.sh
YUP
```

Reply:
In any case, I suggest writing this as `set -o errexit` (it's more explanatory).

Reply:
Your example script does not execute any commands that return a non-zero return code.  The problem is that when:
'''
python example.py
'''
returns an error code, the script exits when `set -e` is used.

Reply:
I think what he means is to write it like 

```
python example.py || echo 'Failed'
```

or something like that.

Reply:
> Your example script does not execute any commands that return a non-zero return code. 
```
$ false
$ echo $?
1
```

Reply:
I guess I am mistaken about `set -e`, but my understanding came from here:
```
(base) Michaels-MBP:PlasCom2 mtcampbe$ help set
set: set [--abefhkmnptuvxBCHP] [-o option] [arg ...]
        -a  Mark variables which are modified or created for export.
        -b  Notify of job termination immediately.
        -e  Exit immediately if a command exits with a non-zero status.
```

Reply:
https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html#The-Set-Builtin

Reply:
Nice, thanks. I think this is fixed up anyway from merged to #55